jormungandr/src/fragment/pool.rs

```diff
@@ -85,12 +65,22 @@ impl Pool {
                         .zip(fragments_exist_in_logs)
                         .filter(|(_, exists_in_logs)| !exists_in_logs)
                         .map(|(fragment, _)| fragment);
-                    let new_fragment_ids = pool.insert_all(new_fragments);
-                    let count = new_fragment_ids.len();
-                    let fragment_logs = new_fragment_ids
-                        .into_iter()
-                        .map(move |id| FragmentLog::new(id.into(), origin));
-                    logs.insert_all(fragment_logs).map(move |_| count)
+                    let new_fragments = pool.insert_all(new_fragments);
+                    let count = new_fragments.len();
+                    let fragment_logs = new_fragments
+                        .iter()
+                        .map(move |fragment| FragmentLog::new(fragment.id().into(), origin))
+                        .collect::<Vec<_>>();
+                    stream::iter_ok(new_fragments)
+                        .map(|fragment| NetworkMsg::Propagate(PropagateMsg::Fragment(fragment)))
+                        .fold(network_msg_box, |network_msg_box, fragment_msg| {
+                            network_msg_box.send(fragment_msg)
+                        })
+                        .map_err(move |err: <MessageBox<_> as Sink>::SinkError| {
```

Comment:
Hmm, interesting that it should need explicit type annotation...

Reply:
I was surprised too